### DISCUSSION TITLE
What's the status on using SWEX with Android ?

### DISCUSSION THREAD

**Discussion by alkanna** [Xzandro/sw-exporter#485] [General]
There was a time when I remember it was possible (2/3 years ago maybe more) to use SWEX to export data from an android phone running an up to date OS.
Right now, I think I tried pretty much everything I could think of on an AOSP android 13 rom, and I feel like there are no good options for android users.

I added the cert to my root store, but it looks like SW is now pinning it directly in okhttp. However they seem to be using a combination of dexguard/appguard in the Android version of the game, I saw a lot of checks regarding emulators, root, xposed and frida.
I tried bypassing the ssl pinning using frida and xposed but that just leads to more problems.

What's the current "official" stance from @Xzandro on that matter at the moment ?

I do personnally have a solution using the native version of SW on my M1 Mac, on which there seems to be no SSL pinning whatsoever (not familiar with iOS developer tools so I don't even know if they allow that), but it is not very practical. I would love to be able to keep the proxy running for my phone which is the main device I play SW on.

**Comment by Xzandro** [comment]
There is no SSL cert pinning at all. The issue lays with Android 7+, because by default, apps do not accept self signed certificates, unless you are putting it into the internal root certificate store, which requires root. There is just nothing we could do about it, so I recommend to use an emulator like Mumu to do it if you only run a mobile device with Android 7+.

**Reply by alkanna** [reply]
It does not work even when putting the cert in the root store of android.

**Reply by alkanna** [reply]
I used to do this to run the proxy in always-on mode on a VM at home, while playing on my phone. It was a Mi Mix 3 at the time and that phone is shipped with Android 9 (Pie). I'm pretty confident there is some sort of pinning taking place because common methods to do just that are being hooked by Frida successfully upon starting the app.

**Reply by Xzandro** [reply]
There is no cert pinning. You just can't proxy the appguard requests, which is why SWEX only proxies the com2us requests itself.
I never tried to root a normal smartphone and put the cert into the root store, since this is also pretty much not practical for normal users and I would never recommend rooting your phone, so I'm just going with the information found online, that in theory, it should work as well.

**Comment by Xzandro** [comment]
Converting to discussion, since this is technically not an issue.

**Comment by alkanna** [comment]
Hello @Xzandro , just checking in if anything has changed in the last few years regarding this ? Still not feasible to run SWEX on a server for use with a real android phone ?

**Reply by Xzandro** [reply]
No, and this will never change until Android changes this policy. Which they probably won't. So in my books, Android phones are pretty much dead for this process.
